fastplotlib/legends/legend.py

```diff
@@ -272,19 +287,74 @@ def add_graphic(self, graphic: Graphic, label: str = None):
 
         graphic.add_event_handler(partial(self.remove_graphic, graphic), "deleted")
 
-        self._col_counter = new_col_ix
-        self._row_counter = new_row_ix
+        self._col_counter = col_idx
+        self._row_counter = row_idx
 
     def _reset_mesh_dims(self):
+
+        # Bounding box of all legend items
         bbox = self._legend_items_group.get_world_bounding_box()
+        (left, bottom, _), (right, top, _) = bbox
+        # width, height, _ = np.ptp(bbox, axis=0)
+
+        pos = self._mesh.geometry.positions.data
+        pos[mesh_masks.x_left] = left - self._padding
+        pos[mesh_masks.x_right] = right + self._padding
+        pos[mesh_masks.y_bottom] = bottom - self._padding
+        pos[mesh_masks.y_top] = top + self._padding
+        self._mesh.geometry.positions.update_range()
+
+    def update_using_camera(self):
+        """
+        Update the legend position and scale using the camera.
+        This only works if legend is in a Dock, not a Plot or Subplot.
+        """
```

Comment:
This will be much easier after #830 is merged (by tomorrow) since it adds an overlay render pass.